### DISCUSSION TITLE
Dependent healthchecks

### DISCUSSION THREAD

**Discussion by alex-todorov-j2** [microsoft/aspire#2651] [Q&A] [answered]
I added a RabbitMQ dependency to the _apiservice_ project:

```
var apiService = builder.AddProject<Projects.AspireStarterApp_ApiService>("apiservice")
    .WithReference(msgBroker);
```

Then stopped the container for RabbitMQ in Docker and went to the /health endpoint of _apiservice_ hoping it would now report Unhealthy. But it still returned Healthy.

My question: is there a way to configure .NET Aspire for nested healthchecks like this? (You could go with your own healthcheck package/code but I'm curious what .NET Aspire has to offer.)

**Comment by DamianEdwards** [comment, marked as answer]
If you use the Aspire Component for RabbitMQ and add the RabbitMQ client using the method provided by it, you should get a health check added for it. See https://learn.microsoft.com/dotnet/aspire/messaging/rabbitmq-client-component?tabs=dotnet-cli#health-checks for more details.

**Reply by alex-todorov-j2** [reply]
True, I was missing the same `var msgBroker = builder.AddRabbitMQ("messaging");` in the apiservice Program class. It is a step in the article. My bad!

With that added, the apiservice /healthcheck returns Unhealthy if RMQ is down. And it reports the same in the Console logs.
